scipp_widgets/input_spec.py

```diff
@@ -59,6 +52,55 @@ def name(self):
         return self._name
 
 
+class StringInputSpec(InputSpecComboboxBase):
+    """
+    Controls creation and validaton of user-input widgets.
+    Processed raw string as input.
+    """
+    def __init__(self,
+                 name,
+                 validator=lambda input: input,
+                 options=(),
+                 tooltip='',
+                 eval_input=False,
+                 scope={}):
+        """
+        Parameters:
+        name (str): Name of function argument this input corresponds to.
+        validator (Callable[[str], Any]): Validator function.
+        options (List[str]): List of dropdown options.
+        tooltip (str): Widget placeholder text.
+        scope (Dict[str: Any]): Non default scope to use for evaluation.
+        """
+        super().__init__(name, options, tooltip, eval_input, scope)
+        self._validator = validator
+
+
+class InputSpec(InputSpecComboboxBase):
+    """
+    Controls creation and validaton of user-input widgets.
+    Evaluates input string in scope
+    """
+    def __init__(self,
+                 name,
+                 validator=lambda input: input,
+                 options=(),
+                 tooltip='',
+                 eval_input=False,
```

Comment:
So with strings out of the way, what is this needed for?

Reply:
No longer needed have removed.